asreview/data/base.py

```diff
@@ -449,3 +449,74 @@ def to_dataframe(self, labels=None, ranking=None):
             result_df.loc[result_df[col_label] == LABEL_NA, col_label] = np.nan
 
         return result_df
+
+    def duplicated(self, pid='doi'):
+        """Create a dataframe with all duplicates based on a custom
+        persistent identifier (PID) and titles/abstracts.
+
+        Arguments
+        ---------
+        pid: string
+            Which persistent identifier to use for deduplication.
+            Default is 'doi'.
+
+        Returns
+        -------
+        pandas.Series
+            Boolean series for each duplicated rows.
+        """
+        if pid in self.df.columns:
+            # in case of strings, strip whitespaces and replace empty strings with None
+            if is_string_dtype(self.df[pid]):
+                s_pid = self.df[pid].str.strip().replace("", None)
+            else:
+                s_pid = self.df[pid]
+
+            # save boolean series for duplicates based on persistent identifiers
+            s_dups_pid = ((s_pid.duplicated()) & (s_pid.notnull()))
+        else:
+            s_dups_pid = None
+
+        # get the texts and clean them
+        s = pd.Series(self.texts) \
+            .str.replace("[^A-Za-z0-9]", "", regex=True) \
+            .str.lower()
+
+        # save boolean series for duplicates based on titles/abstracts
+        s_dups_text = s.duplicated()
+
+        # final boolean series for all duplicates
+        if s_dups_pid is not None:
+            s_dups = np.logical_or(s_dups_pid, s_dups_text)
+        else:
+            s_dups = s_dups_text
+
+        return s_dups
+
+    def drop_duplicates(self, pid='doi', inplace=False, reset_ix=True):
+        """Drop duplicates based on a custom persistent
+        identifier (PID) and titles/abstracts.
+
+        Arguments
+        ---------
+        pid: string, default 'doi'
+            Which persistent identifier to use for deduplication.
+        inplace: boolean, default False
+            Whether to modify the DataFrame rather than creating a new one.
+        reset_ix: boolean, default True
+            If True, the existing index column is reset to the default integer index.
+
+        Returns
+        -------
+        pandas.DataFrame or None
+            DataFrame with duplicates removed or None if inplace=True
+        """
+        if inplace:
+            if reset_ix:
+                self.df = self.df[~self.duplicated(pid)].reset_index(drop=True)
+            else:
+                self.df = self.df[~self.duplicated(pid)]
+        elif reset_ix:
+            return self.df[~self.duplicated(pid)].reset_index(drop=True)
+        else:
+            return self.df[~self.duplicated(pid)]
```

Comment:
```suggestion

        df = self.df[~self.duplicated(pid)]

        if reset_ix:
            df = df.reset_index(drop=True)

        if inplace:
            self.df = df
            return

        return df
```

Same code, slightly easier to understand